### PR TITLE
Improve speed by default buffer size + added parameter for buffer size.

### PR DESCRIPTION
**Pull request by TheCrazyT**
This time without stack overflow.
(I'm wondering why compiler didn't warn about that.)

### CODE REVIEW

File: iqToSharp.cpp [hunk 1]
```diff
@@ -106,15 +111,16 @@ int main(int argc, char **argv){
   fflush(stdout);
   fseek(fout, sizeof(WavHeader), SEEK_SET);
 
-  complex<int8_t> ctx;
+  complex<int8_t> ctx[bufsize];
```
Comment (Marcin648) [inline iqToSharp.cpp]: Instead of allocating a buffer on the stack, use malloc. Remember free this on end of main.

File: makefile [hunk 2]
```diff
@@ -1,7 +1,14 @@
 CXX=g++
 all:
 ifeq ($(OS),Windows_NT)
-	$(CXX) -Wall -Wextra -O3 -static-libstdc++ -static-libgcc -static iqToSharp.cpp -o iqToSharp.exe
+	$(CXX) -Wl,--stack,8388608 -Wall -Wextra -O3 -static-libstdc++ -static-libgcc -static iqToSharp.cpp -o iqToSharp.exe
```
Comment (Marcin648) [inline makefile]: If you use malloc, you don't need to increase stack.

**Review by Marcin648 — CHANGES_REQUESTED** [review]
Thanks for interesting. I don't create buffer because I think that it's easier to someone to pre process samples by adding some own code. But now I think it was a bad idea because it works too slowly.

**Comment by TheCrazyT** [comment]
> 
> 
> Thanks for interesting. I don't create buffer because I think that it's easier to someone to pre process samples by adding some own code. But now I think it was a bad idea because it works too slowly.

You are right about using malloc, I changed the code to use malloc instead.

**Pull request merged**